### ISSUE TITLE
Frontend: Decide on frontend framework

### ISSUE THREAD

**Issue by wknd** [HackThisSite/revival#4]
We gotta decide on what framework to look at.

Considerations:
- most content is static of which the code would be hosted in a different repo
- to save on resources, it is important everything can be pre-rendered

Some options:
- [gatsby](https://www.gatsbyjs.com/) (react based)
- [next.js](https://nextjs.org/) (react based)
- [nuxtJS](https://nuxtjs.org/) (Vue version of next.js)
- [Jekyll](https://jekyllrb.com/) (Used by github, but free build times may be too long for large archives of articles or news) *
- [hugo](https://gohugo.io/) (Written in GO, allegedly fastest build times) *
- [gridsome](https://gridsome.org/) (Vue based)
- [Eleventy](https://www.11ty.dev/) (similar to jekyll?) *
- [scully](https://scully.io/) (Angular based)

(*) frameworks that only build static sites and leave you in full control, they don't add any JS to said static site themselves.

If using one of the framework agnostic builders, we will still need to pick a js framework to handle the dynamic portions.
Some options:
- [React](https://reactjs.org/)
- [Vue](https://vuejs.org/)
- [Solid](https://www.solidjs.com/)
- regular hand crafted javascript

**Comment by simonwuelker** [comment]
https://ssg-build-performance-tests.netlify.app/ might be interesting (gridsome and scully not included)

**Comment by wknd** [comment]
gotta admit currently I am liking hugo with solid.. I'll make a proof of concept for it soon

**Comment by wknd** [comment]
I did a bunch of experiments, and I really like both hugo and solid. But it is not really suited for what I had in mind.

tl;dr: the way solid generates components seemed hard to do dynamically without using javascript itself.
While i got it working fine to have a single login component on every page, it was sorta hacky.
And having for instance a series of components dynamically placed on a page by hugo seemed very hard, you'd have to use js to do it properly.

Astro + solid seemed like it would be better suited for it. Even though its the slowest of the bunch. 
I think astro also recently(or will soon?) added more fancy serverside rendering which could be useful for the more fancy pages.
And for dynamic pages we could also just use solid only.
